broadinstitute/sherlock
# [DDO-3900] Add service banner bucket field to environments

# What
Nicely outlined by Jack in [DDO-3900](https://broadworkbench.atlassian.net/browse/DDO-3900).

TL;DR: Adds a service banner bucket field to environments to allow setting where their Terra UI banner bucket is so we can add more features to set/get/clear the bucket in Sherlock/Beehive later on, replacing the Jenkins banner set/clear jobs, which will allow us to deprecate/shut down Jenkins.

## Details
All environments will have a `service_banner_bucket` field in the db, which can be be `null` or a string representing the name of the GCS bucket where the Terra UI banner files live. If a dynamic environment (BEE) is created from a template, one of the following things will happen:

* If the BEE's template has a service banner bucket set, the BEE will inherit it. I plan to use subdirs within buckets to allow for per-BEE banner files within the same bucket.
* If the BEE's template does not have a service banner bucket set, the BEE won't have a service banner bucket set.

You can change the banner bucket at any time, currently only with the APIs directly, but eventually Beehive will have a metadata field where you will be able to edit it.

# Testing
More tests included in this PR cover the model and API functionality. I also manually did the following to test the new field and functionality:
1. Spin up Sherlock + Postgres locally
2. Create a BEE cluster
3. Create a swatomation template with a default service banner bucket set
4. Create a swatomation template without a default service banner bucket set
5. Create a BEE from (4), the service banner bucket is set, as expected
6. Create a BEE from (5), the service banner bucket is not set, as expected

[DDO-3900]: https://broadworkbench.atlassian.net/browse/DDO-3900?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
